packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,258 @@
+
+export class Drawer extends FASTElement {
+  /**
+   * This method is called when the custom element is connected to the document's DOM.
+   * It overrides the connectedCallback method from the base class.
+   *
+   * @remarks
+   * The connectedCallback method is necessary to ensure that the component is properly initialized and rendered when it is connected to the DOM.
+   * Although the `open` property is initialized to `false` and `openChanged` already calls `show()`, there may be cases where the initial `openChanged` event occurs before the dialog reference (`dialogRef`) is set.
+   * By using the `connectedCallback` method, we can ensure that the component is fully set up before any further actions are taken.
+   */
+  public connectedCallback(): void {
+    super.connectedCallback();
+    Updates.enqueue(() => {
+      this.setComponent();
+    });
+  }
+
+  /**
+   * This method is called when the custom element is disconnected from the document's DOM.
+   */
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+  }
+
+  /**
+   * The dialog element.
+   * @public
+   *
+   */
+  @observable
+  public dialog!: HTMLDialogElement;
+
+  /**
+   * Determines whether the drawer should be displayed as modal, non-modal, or alert.
+   * When in modal or alert mode, an overlay is applied over the rest of the view.
+   * @public
+   * @remarks
+   * HTML Attribute: modalType
+   */
+  @attr({ attribute: 'modal-type' })
+  public modalType: DrawerModalType = DrawerModalType.modal;
+
+  /**
+   * The ID of the element that labels the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-labelledby
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * The ID of the element that describes the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-describedby
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Sets the type of the drawer (overlay/inline).
+   * @public
+   * @remarks
+   * HTML Attribute: type
+   * @defaultValue overlay
+   */
+  @attr
+  public type?: DrawerType;
+
+  /**
+   * Sets the position of the drawer (start/end).
+   * @public
+   * @remarks
+   * HTML Attribute: position
+   * @defaultValue start
+   */
+  @attr
+  public position?: DrawerPosition;
+
+  /**
+   * Sets the size of the drawer (small/medium/large).
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   * @defaultValue medium
+   */
+  @attr({ attribute: 'size' })
+  public size?: DrawerSize;
+
+  /**
+   * Sets the open state of the drawer
+   * @public
+   * @remarks
+   * HTML Attribute: open
+   * @defaultValue false
+   */
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  /**
+   * Indicates whether the drawer is currently closing.
+   */
+  private closing: boolean = false;
+
+  /**
+   * Emits a 'cancel' event.
+   * @internal
+   */
+  public cancel(): void {
```

Comment:
Cancel doesn't seem semantic here to what is happening. This fires when dismissed, why not dismiss?